vmware_aria_operations_integration_sdk/adapter_configurations/adapter_templates/java/sample_adapter/src/main/java/Adapter.java.template

```diff
@@ -1,100 +1,324 @@
+
 package ${package_name};
 
-import com.fasterxml.jackson.databind.JsonNode;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.vmware.aria.operations.*;
+import com.vmware.aria.operations.Object;
+import com.vmware.aria.operations.definition.AdapterDefinition;
+import com.vmware.aria.operations.definition.ObjectType;
+import com.vmware.aria.operations.definition.Units;
+import org.apache.logging.log4j.Logger;
+import oshi.SystemInfo;
+import oshi.hardware.CentralProcessor;
+import oshi.hardware.CentralProcessor.TickType;
 
-import java.io.*;
-import java.util.Arrays;
-import java.util.HashMap;
-import java.util.Map;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.lang.management.OperatingSystemMXBean;
+import java.nio.file.FileStore;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
 
 public class Adapter {
+    private static final String ADAPTER_KIND = "${adapter_key}";
+    private static final String ADAPTER_NAME = "${display_name}" ;
 
-    public static void main(String[] args) {
-        // Check if there are enough command line arguments
-//        ObjectMapper mapper = new ObjectMapper();
-//        ObjectNode rootNode = mapper.createObjectNode();
-//
-//        rootNode.put("name", "John");
-//        rootNode.put("age", 30);
-//
-//        String jsonString = rootNode.toString(); // Convert the JSON object to a string
-//        System.out.println(jsonString);
+    public AdapterDefinition getAdapterDefinition() {
+        Timer timer = new Timer("Get Adapter Definition");
+        Logger logger = AdapterLogger.getLogger();
+        AdapterDefinition definition = null;
+        try {
+            definition = new AdapterDefinition(ADAPTER_KIND, ADAPTER_NAME);
 
-        if (args.length != 3) {
-            System.out.println("Please provide 3 command line arguments.");
-            System.out.println(Arrays.toString(args));
-            return;
-        }
+            definition.defineStringParameter(
+                    "ID",
```

Comment:
We don't need keys to be any particular format, but since we're using snake_case for all the other keys, it'd be good to use `id` for consistency.